pydantic/_internal/_generate_schema.py

```diff
@@ -107,8 +106,7 @@ def check_validator_fields_against_field_name(
 def check_decorator_fields_exist(decorators: Iterable[AnyFieldDecorator], fields: Iterable[str]) -> None:
     fields = set(fields)
     for dec in decorators:
-        if isinstance(dec.info, ValidatorDecoratorInfo) and dec.info.fields == ('*',):
-            # V1 compat: accept `'*'` as a wildcard that matches all fields
+        if isinstance(dec.info, (ValidatorDecoratorInfo, FieldValidatorDecoratorInfo)) and dec.info.fields == ('*',):
```

Comment:
I think better to do `'*' in dec.info.fields` rather than `dec.info.fields == ('*',)`. I could imagine scenarios where someone is deving and adds both `'*'` and a field name, and it ends up pretty confusing that it stops applying to all fields because another field becomes listed. I think this is less of an issue if it errors if you try to do this, I didn't check if it currently does.

Reply:
actually maybe I misunderstood what this check is doing and it's good like this.